internal/extproc/chatcompletion_processor.go

```diff
@@ -227,14 +227,12 @@ func (c *chatCompletionProcessorUpstreamFilter) ProcessRequestHeaders(ctx contex
 
 	// Start tracking metrics for this request.
 	c.metrics.StartRequest(c.requestHeaders)
-	// Response label at this stage defaults to the effective header model; backend override (if any) will be applied in SetBackend.
-	respModel := c.requestHeaders[c.config.modelNameHeaderKey]
-	// Request label should reflect the original user-provided model; fall back to header if body is unavailable.
-	reqModel := respModel
-	if c.originalRequestBody != nil && c.originalRequestBody.Model != "" {
+	// Set the request model for metrics from the original model or override if applied.
+	reqModel := c.requestHeaders[c.config.modelNameHeaderKey]
+	if reqModel == "" && c.originalRequestBody != nil && c.originalRequestBody.Model != "" {
 		reqModel = c.originalRequestBody.Model
```

Comment:
maybe I am missing something but this if block won't be needed. The model override happens within the translation  (`c.translator.translator.RequestBody`) below. I can't think of the situation where `c.originalRequestBody != nil && c.originalRequestBody.Model != ""` is not true so...... 🙇

Reply:
ah sorry nvm